apps/condo/domains/billing/utils/validation.utils.js

```diff
@@ -103,18 +103,40 @@ const REPORT_SCHEMA = {
 
 const _jsonReportValidator = ajv.compile(REPORT_SCHEMA)
 
-function _validateReport ({ resolvedData, fieldPath, addFieldValidationError }) {
+function validateReport ({ resolvedData, fieldPath, addFieldValidationError }) {
     if (!_jsonReportValidator(resolvedData[fieldPath])) {
         return _jsonReportValidator.errors.forEach((error) => {
             addFieldValidationError(`${fieldPath} field validation error. JSON not in the correct format - path:${error.instancePath} msg:${error.message}`)
         })
     }
 }
 
+const DATA_FORMAT_SCHEMA = {
+    type: 'object',
+    properties: {
+        hasToPayDetail: { type: 'boolean' },
+        hasServices: { type: 'boolean' },
+        hasServicesDetail: { type: 'boolean' },
+    },
+    required: ['hasToPayDetail', 'hasServices', 'hasServicesDetail'],
+    additionalProperties: false,
+}
+
+const _jsonDataFormatValidator = ajv.compile(DATA_FORMAT_SCHEMA)
+
+function validateDataFormat ({ resolvedData, fieldPath, addFieldValidationError }) {
+    if (!_jsonDataFormatValidator(resolvedData[fieldPath])) {
+        return _jsonDataFormatValidator.errors.forEach((error) => {
```

Comment:
Return void? Not sure this is correct, at least not obvious.

Reply:
We actually need to just call `addValidationError` to add the error, so we don't need to return the objects with validation errors

I agree on a point that this may not be obvious, but this works correctly – you can check the proof in `tests`

Reply:
I just removed the return to make this lad more obvious